### PR TITLE
Storyfeed Updates

### PR DESCRIPTION
**Pull request by nickreynolds**
- replace "boosts" tab on listing page with "storyfeed", default to storyfeed for whitelisted newsrooms
- refactor so boost feed uses same component and just passes different filter

### CODE REVIEW

File: packages/dapp/src/components/Main.tsx [hunk 1]
```diff
@@ -158,6 +162,11 @@ export const Main: React.FunctionComponent = () => {
           <Route path={routes.SUBMIT_CHALLENGE} component={AsyncComponent(SubmitChallengePage)} />
           <Route path={routes.SUBMIT_APPEAL_CHALLENGE} component={AsyncComponent(SubmitAppealChallengePage)} />
           <Route path={routes.REQUEST_APPEAL} component={AsyncComponent(RequestAppealPage)} />
+          <Route path={routes.LISTING_STORY_BOOST_PAYMENT} component={AsyncComponent(Listing, { payment: true })} />
+          <Route
+            path={routes.LISTING_STORY_BOOST_NEWSROOM}
+            component={AsyncComponent(Listing, { newsroomDetails: true })}
+          />
```
Comment (sruddy) [inline packages/dapp/src/components/Main.tsx]: If the newsroom modal is blocked from rendering on listing page you don't need this.

File: packages/dapp/src/constants.ts [hunk 2]
```diff
@@ -8,7 +8,9 @@ export enum routes {
   REGISTRY_HOME_ROOT = "/registry",
   HOMEPAGE = "/",
   CONTRACT_ADDRESSES = "/contract-addresses",
-  LISTING = "/listing/:listingAddress/:activeTab?",
+  LISTING = "/listing/:listingAddress/:activeTab?/:postId?",
+  LISTING_STORY_BOOST_PAYMENT = "/listing/:listingAddress/:activeTab?/:postId/payment",
+  LISTING_STORY_BOOST_NEWSROOM = "/listing/:listingAddress/:activeTab?/:postId/newsroom",
```
Comment (sruddy) [inline packages/dapp/src/constants.ts]: Same here - If the newsroom modal is blocked from rendering on listing page you don't need this.

File: packages/dapp/src/components/listing/ListingRedux.tsx [hunk 3]
```diff
@@ -221,6 +241,38 @@ class ListingPageComponent extends React.Component<
...
+    });
+  };
+
+  private openStoryNewsroomDetails = (postId: string) => {
+    let urlBase = this.props.location.pathname;
+    urlBase = urlBase.substring(0, urlBase.indexOf("/"));
+    this.props.history.push({
+      pathname: urlBase + "/listing/" + this.props.listingId + "/storyfeed/" + postId + "/newsroom",
+    });
+  };
```
Comment (sruddy) [inline packages/dapp/src/components/listing/ListingRedux.tsx]: Same here - If the newsroom modal is blocked from rendering on listing page you don't need this.

File: packages/dapp/src/components/listing/Listing.tsx [hunk 4]
```diff
@@ -31,6 +33,7 @@ export interface PreListingReduxProps {
 
 class ListingPageComponent extends React.Component<ListingPageProps> {
   public render(): JSX.Element {
+    console.log("ListingPage payment?: ", this.props.payment);
```
Comment (sruddy) [inline packages/dapp/src/components/listing/Listing.tsx]: extra console.log

File: packages/dapp/src/components/listing/ListingRedux.tsx [hunk 5]
```diff
@@ -108,17 +112,25 @@ class ListingPageComponent extends React.Component<
...
       activeTabIndex = TABS.indexOf(activeTab) || 0;
+    } else if (listing && listing.data.isWhitelisted) {
+      activeTabIndex = 0;
     } else if (listing && listing.data.challenge) {
+      activeTabIndex = 2;
+    } else {
       activeTabIndex = 1;
     }
 
+    console.log("Set Active Tab Index: ", activeTabIndex);
```
Comment (sruddy) [inline packages/dapp/src/components/listing/ListingRedux.tsx]: extra console.log

File: packages/dapp/src/components/listing/ListingRedux.tsx [hunk 6]
```diff
@@ -108,17 +112,25 @@ class ListingPageComponent extends React.Component<
...
+
     this.setState({ activeTabIndex });
   }
 
   public render(): JSX.Element {
     const { listing, newsroom, listingPhaseState, charterRevisionId, charterRevisions } = this.props;
     const listingExistsAsNewsroom = listing && newsroom;
 
+    console.log("payment?: ", this.props.payment);
```
Comment (sruddy) [inline packages/dapp/src/components/listing/ListingRedux.tsx]: extra console.log

File: packages/dapp/src/components/StoryFeed/StoryFeed.tsx [hunk 7]
```diff
@@ -25,156 +30,149 @@ class StoryFeedPage extends React.Component<StoryFeedPageProps> {
...
-  private closeStoryBoost = () => {
-    let urlBase = this.props.location.pathname;
-    urlBase = urlBase.substring(0, urlBase.indexOf("/"));
-    this.props.history.push({
-      pathname: urlBase + "/storyfeed",
-    });
-  };
-
-  private openStoryDetails = (postId: string) => {
-    this.context.fireAnalyticsEvent("story boost", "story details clicked", postId);
```
Comment (sruddy) [inline packages/dapp/src/components/StoryFeed/StoryFeed.tsx]: Are these GA events added back somewhere?

File: packages/dapp/src/components/StoryFeed/StoryFeed.tsx [hunk 8]
```diff
@@ -25,156 +30,149 @@ class StoryFeedPage extends React.Component<StoryFeedPageProps> {
...
-    let urlBase = this.props.location.pathname;
-    urlBase = urlBase.substring(0, urlBase.indexOf("/"));
-    this.props.history.push({
-      pathname: urlBase + "/storyfeed/" + postId,
-    });
-  };
-
-  private openPayments = (postId: string) => {
-    this.context.fireAnalyticsEvent("story boost", "boost button clicked", postId);
```
Comment (sruddy) [inline packages/dapp/src/components/StoryFeed/StoryFeed.tsx]: GA tracking

File: packages/dapp/src/components/StoryFeed/StoryFeed.tsx [hunk 9]
```diff
@@ -25,156 +30,149 @@ class StoryFeedPage extends React.Component<StoryFeedPageProps> {
...
-    let urlBase = this.props.location.pathname;
-    urlBase = urlBase.substring(0, urlBase.indexOf("/"));
-    this.props.history.push({
-      pathname: urlBase + "/storyfeed/" + postId + "/payment",
-    });
-  };
-
-  private openStoryNewsroomDetails = (postId: string) => {
-    this.context.fireAnalyticsEvent("story boost", "newsroom details clicked", postId);
```
Comment (sruddy) [inline packages/dapp/src/components/StoryFeed/StoryFeed.tsx]: GA tracking

**Comment by dankins** [comment]
:tada: This PR is included in version 4.9.8 :tada:

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

**Pull request merged**